utils/analysis/filter.go

```diff
@@ -115,6 +115,9 @@ func FilterAnalysisRunsToDelete(ars []*v1alpha1.AnalysisRun, olderRSs []*appsv1.
 	olderRsPodHashes := map[string]bool{}
 	for i := range olderRSs {
 		rs := olderRSs[i]
+		if rs == nil {
```

Comment:
Just curious how this happens and if theres a bug somewhere, and we are addressing the symptom and not the problem. Worried that other parts of code expect olderRSs to always have valid ReplicaSets.

Reply:
Discussed IRL, but the newRS can be nil if there is a scaling event and a spec change